compressor/Bit_cashier.cpp

```diff
@@ -0,0 +1,65 @@
+#include <iostream>
+#include <deque>
+#include <string>
+
+class Bit_cashier
+{
+  std::deque<bool> bits;
+  const int byte_length = 8;
+
+  void put_error_and_exit(const std::string& message)
+  {
+    std::cerr << message;
+    exit(1);
+  }
+
+public:
+  Bit_cashier(){}
+
+  void add(const bool new_value)
+  {
+    bits.push_back(new_value);
+  }
+
+  void add(const char future_bits)  
```

Comment:
char usually is a signed int. Hence usually it's only 7 bits and sign. For 8 bits you better use unsigned char. And in C++ 2011 standart you can use uint8_t which is even better.
http://en.cppreference.com/w/cpp/types/integer